frontend/src/app/components/filters/filter-searchable-multiselect-value/filter-searchable-multiselect-value.component.ts

```diff
@@ -0,0 +1,154 @@
+import { APIV3WorkPackagePaths } from 'core-app/modules/apiv3/endpoints/work_packages/api-v3-work-package-paths';
+import { WorkPackageResource } from 'core-app/modules/hal/resources/work-package-resource';
+
+export interface FilterConditions {name:string; operator:FilterOperator; values:unknown[]|boolean; }
+
+@Component({
+  selector: 'filter-searchable-multiselect-value',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  templateUrl:'./filter-searchable-multiselect-value.component.html'
+})
+
+
+export class FilterSearchableMultiselectValueComponent implements OnInit {
+  @Input() public shouldFocus:boolean = false;
+  @Input() public filter:QueryFilterInstanceResource;
+  @Input() public filterConditions?:FilterConditions[];
+  @Input('filterResource')public filterResource:'work_packages' | 'users';
+  @Input() public filterSearchKey?:string;
+  @Output() public filterChanged = new EventEmitter<QueryFilterInstanceResource>();
+
+  @ViewChild('ngSelectInstance', { static: true }) ngSelectInstance:NgSelectComponent;
+
+  public _availableOptions:HalResource[] = [];
+  public compareByHrefOrString = AngularTrackingHelpers.compareByHrefOrString;
+
+  private _isEmpty:boolean;
+
+  readonly text = {
+    placeholder: this.I18n.t('js.placeholders.selection'),
+  };
+
+   constructor(readonly halResourceService:HalResourceService,
+              readonly halSorting:HalResourceSortingService,
+              readonly apiV3Service:APIV3Service,
+              readonly cdRef:ChangeDetectorRef,
+              readonly I18n:I18nService,
+              protected currentProject:CurrentProjectService,
+              readonly halNotification:HalResourceNotificationService) {
+  }
+  public requests = new DebouncedRequestSwitchmap<string, ValueOption>(
+    (searchTerm:string) => this.loadAvailable(searchTerm),
+    errorNotificationHandler(this.halNotification),
+    true
+  );
+  public active:Set<string>;
+
+  protected createFilters(filterConditions:FilterConditions[], matching:string) {
+    let filters = new ApiV3FilterBuilder();
+    for (let condition of filterConditions) {
+      filters.add(condition.name, condition.operator, condition.values);
+    }
+    if (matching) {
+      filters.add(this.filterSearchKey ?? '', '**', [matching]);
+    }
+    return filters;
+  }
+  public loadAvailable(matching:string):Observable<HalResource[]> {
+
+    let filters:ApiV3FilterBuilder = this.createFilters(this.filterConditions ?? [], matching);
+
+    let filteredData = (this.apiV3Service[this.filterResource] as
+      APIv3ResourceCollection<UserResource|WorkPackageResource,
+      APIv3UserPaths|APIV3WorkPackagePaths>).filtered(filters)
+      .get()
+      .pipe(
+        map(collection => collection.elements)
+      );
+
+    return filteredData
+      .pipe(
+        map(items => items
+      ));
+  }
+
+  initialization() {
+    this
+    .requests
+    .output$
```

Comment:
Most of the time, when we subscribe to an observable, we need to unsubscribe when the component is destroyed. There are a few ways of doing so but in this project, the usual is to extend the UntilDestroyedMixin and then pipe the observable like this .pipe(this.untilDestroyed()).

Not unsubscribing would lead to memory leaks. An example of an observable that doesn't need unsubscription is an HTTP call because HTTP calls return a value and then complete.